src/test/ui/diagnostic-item.rs

```diff
@@ -0,0 +1,10 @@
+#[rustc_diagnostic_item = "ok"]
+struct Hello;
+
+#[rustc_diagnostic_item] //~ ERROR expected a string value
+#[rustc_diagnostic_item = 7] //~ ERROR expected a string value
+trait Fam {}
+
+#[rustc_diagnostic_item = "too"]
+#[rustc_diagnostic_item = "much"] //~ ERROR multiple rustc_diagnostic_item attributes found
```

Comment:
you need to run `./x.py test src/test/ui --bless --test-args diagnostic-items` in order to generate the corresponding `.stderr` file

Reply:
Thanks. Does `//~ ERROR..` do anything?

Reply:
It's just an error annotation, you have to run tests with the `--bless` to generate `.stderr` file (https://rustc-dev-guide.rust-lang.org/tests/running.html#editing-and-updating-the-reference-files may be helpful).

Reply:
yes, without that the test will fail because you got "unexpected errors". All errors marked with that comment will be "expected errors" and thus fail the test if they don't happen